shared/fs/index.js

```diff
@@ -1,30 +1,110 @@
 // @flow
-import * as React from 'react'
-import {globalStyles, globalMargins} from '../styles'
-import {Box, Button, Text, ButtonBar} from '../common-adapters'
-
-type Props = {
-  counter: number,
-  you: ?string,
-  increase: () => void,
-  increase10: () => void,
+import React, {PureComponent} from 'react'
+import {FolderVisibility} from '../constants/types/fs'
+import {globalStyles, globalColors, globalMargins} from '../styles'
+import {isMobile} from '../constants/platform'
+import {Box, Button, ClickableBox, Icon, List, Text} from '../common-adapters'
+import {type IconType} from '../common-adapters/icon'
+
+const stylesCommonCore = {
+  alignItems: 'center',
+  borderBottomColor: globalColors.black_05,
+  borderBottomWidth: 1,
+  justifyContent: 'center',
 }
 
-const Fs = ({counter, you, increase, increase10}: Props) => (
-  <Box style={containerStyle}>
-    <Text type="Header">Hi {you}!</Text>
-    <Text type="BodySemibold">Count: {counter}</Text>
-    <ButtonBar align="flex-start">
-      <Button type="Primary" onClick={increase} label="Up by 1" />
-      <Button type="Primary" onClick={increase10} label="Up by 10" />
-    </ButtonBar>
-  </Box>
-)
+const stylesCommonRow = {
+  ...globalStyles.flexBoxRow,
+  ...stylesCommonCore,
+  minHeight: isMobile ? 64 : 48,
+  padding: 8,
+}
 
-const containerStyle = {
+const stylesContainer = {
   ...globalStyles.flexBoxColumn,
+  ...globalStyles.fullHeight,
   flex: 1,
-  padding: globalMargins.xlarge,
 }
 
+const rootFolders = [
+  'private',
+  'public',
+  'team',
+]
+
+type FolderHeaderProps = {
+  title: string,
+}
+
+type FileRowProps = {
+  path: string,
+  icon: IconType,
+  showFileData: () => void,
+}
+
+type FolderProps = {
+  path: string,
+  visibility: FolderVisibility,
+  items: Array<string>,
+}
+
+const FolderHeader = ({title}: FolderHeaderProps) => (
+  <Box>
+    <Box style={{...stylesCommonRow, alignItems: 'center', borderBottomWidth: 0}}>
+      <Text type="HeaderBig" style={{padding: globalMargins.xtiny}}>
+        {title}
+      </Text>
+    </Box>
+  </Box>
+)
+
+
+const FileRow = ({path, icon, showFileData}: FileRowProps) => (
+  <ClickableBox onClick={showFileData} style={{...stylesCommonRow}}>
+    <Box key={path} style={{...globalStyles.flexBoxRow, alignItems: 'center', flex: 1}}>
+      <Icon type={icon} style={{marginRight: 16}} />
+      <Box style={{...globalStyles.flexBoxColumn, flex: 1, justifyContent: 'flex-start'}}>
+        <Text type="Body" style={{flex: 1}}>
+          {path}
+        </Text>
+      </Box>
+    </Box>
+  </ClickableBox>
+)
+
+class Folder extends PureComponent<FolderProps> {
+  _renderRow(index, item) {
+    return (
+      <FileRow path={item.path} icon={item.icon} showFileData={item.showFileData} />
+    )
+  }
+
+  render() {
+    const iconTypes : IconType = {
+      folder: isMobile ? 'icon-folder-private-24' : 'icon-folder-private-24',
+      file: isMobile ? 'icon-file-24' : 'icon-file-24',
+    }
+    const items = [
```

Comment:
This form will cause a lot of render thrashing as whats passed down to <List will be new every time (even if nothing changes). Ideally the connect would return just an array of ids thats passed down and each row would be connected. 
Since this is a simple skeleton its fine to not have this be connected right now, but still you likely want to just pass this.props.items into <List directly and have _renderRow do this kind of map (the item passed there is essentially what you get on line 88 here)